typescript/src/isInBannedPosition.ts

```diff
@@ -0,0 +1,17 @@
+import type tslib from 'typescript/lib/tsserverlibrary'
+
+export default (
+    position: number,
+    fileName: string,
+    scriptSnapshot: tslib.IScriptSnapshot,
+    sourceFile: tslib.SourceFile,
+    languageService: tslib.LanguageService,
+): boolean => {
+    const { character } = languageService.toLineColumnOffset!(fileName, position)
+    const textBeforePositionLine = scriptSnapshot?.getText(position - character, position)
+    const textAfterPosition = scriptSnapshot?.getText(position, sourceFile.getLineEndOfPosition(position))
+    const newLineIndex = textAfterPosition.indexOf('\n')
```

Comment:
We already picked text above with `getLineEndOfPosition`, there is no need to seek for the end of line!